api/host.py

```diff
@@ -226,6 +226,38 @@ def _delete_filtered_hosts(host_id_list):
     return deletion_count
 
 
+@api_operation
+@rbac(Permission.WRITE)
+@metrics.api_request_time.time()
+def delete_all_hosts(delete_all=None, confirm_delete_all=None):
+    if not any([delete_all, confirm_delete_all]):
+        logger.error("To delete all hosts, provide delete_all=true and confirm_delete_all=true in the request.")
+        flask.abort(400, "To delete all hosts, provide delete_all=true and confirm_delete_all=true in the request.")
+    if (delete_all and not confirm_delete_all) or (confirm_delete_all and not delete_all):
+        logger.error("Deleting all hosts requires delete_all=true and confirm_delete_all=true.")
+        flask.abort(400, "Deleting all hosts requires delete_all=true and confirm_delete_all=true.")
```

Comment:
I don't think it's really necessary to have two separate errors for sending none and sending one.
```suggestion
    if not all([delete_all, confirm_delete_all]):
        logger.error("To delete all hosts, provide delete_all=true and confirm_delete_all=true in the request.")
        flask.abort(400, "To delete all hosts, provide delete_all=true and confirm_delete_all=true in the request.")
```

Reply:
Yes. I'm suggesting we just return one error message if both or either is missing, and `if not all([delete_all, confirm_delete_all])` will cover all of those cases.

Also, do we really need the `delete_all_hosts=true` parameter if that's the sole purpose of the endpoint? I'm still all for requiring confirmation, but I think `confirm=true` is enough. Originally we only had `delete_all_hosts` to tell the api what we were trying to do since it was part of the filtered host delete endpoint but it feels superfluous here.

Reply:
Yes, earlier missed `all`.  Changed it to use `if not all([delete_all, confirm_delete_all]):`